api_tests/preprints/views/test_preprint_list_mixin.py

```diff
@@ -73,24 +73,59 @@ def test_filter_published_false_public(self):
         res = self.app.get('{}filter[is_published]=false'.format(self.url))
         assert len(res.json['data']) == 0
 
+@pytest.mark.django_db
 class PreprintIsValidListMixin(object):
 
-    def setUp(self):
-        super(PreprintIsValidListMixin, self).setUp()
-        assert self.admin, 'Subclasses of PreprintIsValidListMixin must define self.admin'
-        assert self.project, 'Subclasses of PreprintIsValidListMixin must define self.project'
-        assert self.provider, 'Subclasses of PreprintIsValidListMixin must define self.provider'
-        assert self.url, 'Subclasses of PreprintIsValidListMixin must define self.url'
+    # FIXTURES
+    @pytest.fixture()
+    def admin(self):
+        raise NotImplementedError("subclass must define an admin fixture")
+
+    @pytest.fixture()
+    def url(self):
+        raise NotImplementedError("subclass must define a url fixture")
+
+    @pytest.fixture()
+    def project(self, admin):
+        raise NotImplementedError("subclass must define a project fixture")
+
+    @pytest.fixture()
+    def provider(self):
+        raise NotImplementedError("subclass must define a provider")
+
+    @pytest.fixture()
+    def app(self):
+        return JSONAPITestApp()
+
+    @pytest.fixture()
+    def write_contrib(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def non_contrib(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def subject(self):
+        return SubjectFactory()
+
+    # SETUP
+    @pytest.fixture(scope="function", autouse=True)
+    def int(self, app, url, admin, write_contrib, non_contrib, project, provider, subject):
+        self.app = app
+        self.url = url
+        self.admin = admin
+        self.write_contrib = write_contrib
+        self.non_contrib = non_contrib
+        self.project = project
+        self.provider = provider
+        self.subject = subject
 
-        self.write_contrib = AuthUserFactory()
-        self.non_contrib = AuthUserFactory()
-
-        self.subject = SubjectFactory()
-        self.file_one_project = test_utils.create_test_file(self.project, self.admin, 'saor.pdf')
         self.project.add_contributor(self.write_contrib, permissions=permissions.DEFAULT_CONTRIBUTOR_PERMISSIONS, save=True)
+        test_utils.create_test_file(self.project, self.admin, 'saor.pdf')
         self.preprint = PreprintFactory(creator=self.admin, filename='saor.pdf', provider=self.provider, subjects=[[self.subject._id]], project=self.project, is_published=True)
 
-    # Test private
+    # TESTS
```

Comment:
I think the comment this replaces was accidental. Probably can remove this one too